packages/proxy/scripts/sync_models.ts

```diff
@@ -0,0 +1,677 @@
+          } catch (error) {
+            if (error instanceof z.ZodError) {
+              console.error(
+                "Zod validation errors in remote data:",
+                error.errors,
+              );
+              reject(
+                new Error(
+                  "Failed to parse remote JSON due to schema validation errors.",
+                ),
+              );
+            } else {
+              reject(
+                new Error(
+                  "Failed to parse remote JSON: " + (error as Error).message,
+                ),
+              );
+            }
+          }
+        });
+      })
+      .on("error", (err) => {
+        reject(new Error("Failed to fetch remote models: " + err.message));
+      });
+  });
+}
+
+async function readLocalModels(filePath: string): Promise<LocalModelList> {
+  try {
+    const fileContent = await fs.promises.readFile(filePath, "utf-8");
+    const localData = JSON.parse(fileContent);
+    // Validate local data with the imported ModelSchema
+    return z.record(ModelSchema).parse(localData);
+  } catch (error) {
+    if (error instanceof z.ZodError) {
+      console.error(
+        "Zod validation errors in local model_list.json:",
+        error.errors,
+      );
+      throw new Error("Local model_list.json failed Zod validation.");
+    }
+    throw new Error(
+      "Failed to read or parse local model list: " + (error as Error).message,
+    );
+  }
+}
+
+function translateLiteLLMToBraintrust(
+  modelName: string,
+  provider?: string,
+): string {
+  if (provider === "xai" && modelName.startsWith("xai/")) {
+    return modelName.substring(4); // "xai/"
+  }
+
+  if (provider === "gemini") {
+    if (modelName.startsWith("gemini/gemini-gemma-")) {
+      return "google/" + modelName.substring(14);
+    }
+    if (modelName.startsWith("gemini/gemma-")) {
+      return "google/" + modelName.substring(7);
+    }
+    if (modelName.startsWith("gemini/")) {
+      return modelName.substring(7);
+    }
+  }
+
+  if (modelName.startsWith("google/")) {
+    return modelName;
+  }
+
+  return modelName;
+}
+
+async function findMissingCommand(argv: any) {
+  try {
+    console.log("Fetching remote models from:", REMOTE_MODEL_URL);
+    const remoteModels = await fetchRemoteModels(REMOTE_MODEL_URL);
+    console.log(`Fetched ${Object.keys(remoteModels).length} remote models.`);
+
+    console.log("Reading local models from:", LOCAL_MODEL_LIST_PATH);
+    const localModels = await readLocalModels(LOCAL_MODEL_LIST_PATH);
+    console.log(`Read ${Object.keys(localModels).length} local models.`);
+
+    const localModelNames = new Set(Object.keys(localModels));
+    const missingInLocal: string[] = [];
+    const consideredRemoteModels: LiteLLMModelList = {};
+
+    for (const remoteModelName in remoteModels) {
+      const modelDetail = remoteModels[remoteModelName];
+
+      if (argv.provider) {
+        const lowerArgProvider = argv.provider.toLowerCase();
```

Comment:
#nit "provider" (and lowerArgProvider) are a bit vague/indirect

may be more readable if it we wrote the code as `searchProvider`

Reply:
I think i prefer to keep as is for literal argument parsing